bin/devtools.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:http_server/http_server.dart' show VirtualDirectory;
+import 'package:path/path.dart';
+import 'package:args/args.dart';
+
+const argHelp = 'help';
+const argMachine = 'machine';
+const argPort = 'port';
+const startPort = 8123;
+
+final argParser = new ArgParser()
+  ..addFlag(argHelp, hide: true)
+  ..addFlag(argMachine,
+      negatable: false,
+      abbr: 'm',
+      help: 'Sets output format to JSON for consumption in tools')
+  ..addOption(argPort, abbr: 'p', help: 'Port to serve the web application on');
```

Comment:
default to something like `9100`?

Reply:
If we add a default, I think we'd have to either:

- Crash if it's already taken
- Try a different port if it's already taken

I think the first adds an unnecessary burden to the client (to handle this) and the second seems to make the default a little pointless.

Is there a significant advantage to a default? I don't expect anyone to ever be typing this manually (and if they did, they should specify the one they want).

Reply:
I think failing to start the the app (w/ a good error message) is reasonable (`Could not start Dart DevTools on port xxx.`)

An IDE client can definitely either choose an explicit port to start it on, or ask for port `0` and then listen on stdout for the port that was selected - it won't be important for those clients to have a consistent port number.

It will be nice for people using this from the cli - its a small thing, but having a small, easy to remember port is a nice feature.

It's not a blocker for this PR, but something we may want to change later.

Reply:
>  or ask for port `0`

Aha, I hadn't thought about that. I think this addresses my concern :-)

I'll make a note to fix in the morning, but I'll land this in the meantime in case you want to progress a package.